app/models/project.rb

```diff
@@ -750,6 +750,29 @@ def mailing_list(include_prereleases: false)
     subs.reject { |sub| mutes.include?(sub.id) }
   end
 
+  # @return [Array(String)] All possible Version#repository_sources for
+  #                         this package, unordered
+  def repository_sources
+    versions
+      .flat_map(&:repository_sources)
+      .compact
+      .uniq
+  end
+
+  # @return [Version,nil] If version was provided, the specific Version or
+  #                       nil if not found. If not provided, the most
+  #                       recently created Version.
```

Comment:
Should this be `most_recently_created_version` or `most_recently_published_version`? Alternatively, worth splitting up into two methods so that one method isn't doing too much?

Reply:
The original code used most recently created, so I'm not changing the calculation for that.